File: VB/WindowsApplication73/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid

Namespace ImmediateInitRow

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub gridView1_InitNewRow(ByVal sender As Object, ByVal e As InitNewRowEventArgs)
            Dim view As GridView = TryCast(sender, GridView)
            view.SetRowCellValue(e.RowHandle, view.Columns("Name"), "Noname")
            view.SetRowCellValue(e.RowHandle, view.Columns("Date"), Date.Today)
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
        End Sub

        Private Sub gridView1_ShownEditor(ByVal sender As Object, ByVal e As EventArgs)
            Dim view As GridView = TryCast(sender, GridView)
            If view.IsNewItemRow(view.FocusedRowHandle) AndAlso Not Equals(view.FocusedColumn.FieldName, "ID") Then view.ActiveEditor.IsModified = True
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication73/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;

namespace ImmediateInitRow
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void gridView1_InitNewRow(object sender, DevExpress.XtraGrid.Views.Grid.InitNewRowEventArgs e)
        {

             GridView view = sender as GridView;
             view.SetRowCellValue(e.RowHandle, view.Columns["Name"], "Noname");
             view.SetRowCellValue(e.RowHandle, view.Columns["Date"], DateTime.Today);
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void gridView1_ShownEditor(object sender, EventArgs e)
        {
            GridView view = sender as GridView;
            if (view.IsNewItemRow(view.FocusedRowHandle) && view.FocusedColumn.FieldName != "ID")
                view.ActiveEditor.IsModified = true;
        }
    }
}